owners-bot/test/index.test.js

```diff
@@ -0,0 +1,512 @@
+const rerequest35 = require('./fixtures/actions/rerequested.35');
+const review35 = require('./fixtures/actions/pull_request_review.35.submitted');
+
+const files35 = require('./fixtures/files/files.35');
+const files35Multiple = require('./fixtures/files/files.35.multiple');
+const files36 = require('./fixtures/files/files.36');
+
+const reviews35 = require('./fixtures/reviews/reviews.35');
+const reviews35Approved = require('./fixtures/reviews/reviews.35.approved');
+
+const checkruns35 = require('./fixtures/check-runs/check-runs.get.35');
+const checkruns35Multiple = require('./fixtures/check-runs/check-runs.get.35.multiple');
+const checkruns35Empty = require('./fixtures/check-runs/check-runs.get.35.empty');
+const checkRunsCreate = require('./fixtures/check-runs/check-runs.create');
+
+const pullRequest35 = require('./fixtures/pulls/pull_request.35');
+
+const teams = require('./fixtures/teams/teams');
+
+
+nock.disableNetConnect();
+jest.setTimeout(30000);
+
+const ownersYamlStruct = {
+  '.': {
+    'path': './OWNERS.yaml',
+    'dirname': '.',
+    'fullpath': '/Users/erwinm/dev/github-owners-bot-test-repo/OWNERS.yaml',
+    'score': 0,
+    'dirOwners': [
+      'donttrustthisbot'
+    ],
+    'fileOwners': {}
+  },
+  './dir1': {
+    'path': './dir1/OWNERS.yaml',
+    'dirname': './dir1',
+    'fullpath': '/Users/erwinm/dev/github-owners-bot-test-repo/dir1/OWNERS.yaml',
+    'score': 1,
+    'dirOwners': [
+      'donttrustthisbot'
+    ],
+    'fileOwners': {}
+  },
+  './dir2': {
+    'path': './dir2/OWNERS.yaml',
+    'dirname': './dir2',
+    'fullpath': '/Users/erwinm/dev/github-owners-bot-test-repo/dir2/OWNERS.yaml',
+    'score': 1,
+    'dirOwners': [
+      'erwinmombay'
+    ],
+    'fileOwners': {}
+  },
+  './dir2/dir1/dir1': {
+    'path': './dir2/dir1/dir1/OWNERS.yaml',
+    'dirname': './dir2/dir1/dir1',
+    'fullpath': '/Users/erwinm/dev/github-owners-bot-test-repo/dir2/dir1/dir1/OWNERS.yaml',
+    'score': 3,
+    'dirOwners': [
+      'erwinmombay'
+    ],
+    'fileOwners': {}
+  }
+};
+
+describe('owners bot', () => {
+  let probot;
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+    sandbox.stub(Git.prototype, 'pullLatestForRepo').returns(null);
+    sandbox.stub(Git.prototype, 'getOwnersFilesForBranch')
+        .returns(ownersYamlStruct);
+
+    probot = new Probot({})
+    const app = probot.load(owners);
+
+    // just return a test token
+    app.app = () => 'test';
+  })
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('when there are more than 1 checks on a PR', () => {
+
+    test('it should update amp owners bot check when there is one', async () => {
+
+      nock('https://api.github.com')
+        .post('/app/installations/588033/access_tokens')
+        .reply(200, {token: 'test'});
+
+      // We need the list of files on a pull request to evaluate the required
+      // reviewers.
+      nock('https://api.github.com')
```

Comment:
Consider adding nocks expectations like I did here:
https://github.com/ampproject/amp-github-apps/blob/master/bundle-size/test/app.test.js#L324-L348

You can chain multiple definitions together (in the above example `.get()` and `.patch()` are chained, then calling `waitUntilNockScopeIsDone(nocks)` (defined https://github.com/ampproject/amp-github-apps/blob/master/bundle-size/test/app.test.js#L52-L58) ensures that all of those networks mock were actually called